c/src/ml-api-inference-pipeline.c

```diff
@@ -1789,31 +1783,43 @@ ml_pipeline_src_input_data (ml_pipeline_src_h h, ml_tensors_data_h data,
   handle_exit (h);
 }
 
+/**
+ * @brief Internal function to fetch ml_pipeline_src_callbacks_s pointer
+ */
+static ml_pipeline_src_callbacks_s *
+get_app_src_callback (ml_pipeline_common_elem * src_h, void **data)
+{
+  ml_pipeline_src_callbacks_s *src_cb = NULL;
+  ml_pipeline_element *elem;
+
```

Comment:
Don't we need to check parameter?
```
g_assert(src_h);
g_assert(data);
```

Reply:
I agreed with your opinion. Assertion macro is necessary before processing internal logic. However, ML API is released without debug setting and it could cause abnormal application termination.

In my opinion, we need to talk about this issue first before applying the assertion macro. For example, the macro can be turned off in the product release by defining `G_DISABLE_ASSERT` when compiling the application. What about applying this patch first?

Reply:
I agree with you. Wihout release mode, `assert` may work differently from expectation.